File: packages/elixir-sdk/test/adapters/migrations_test.exs
defmodule TreeDxSdk.MigrationsAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Migrations.create(client, "repo/a", %{})
    TreeDxSdk.Migrations.get(client, "repo/a", "mig/a")

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :get and &1.path == "/api/v1/repos/repo%2Fa/migrations/mig%2Fa")
           )
  end
end

File: packages/ts-sdk/test/adapters/storage/migrations.test.ts
import { describe, expect, it } from 'vitest';
import { MigrationsAdapter } from '../../../src/treedx/index.js';
import { MockTransport } from '../mock.js';

describe('MigrationsAdapter', () => {
  it('constructs migration endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new MigrationsAdapter({ transport });
    await adapter.create('repo', {});
    await adapter.get('repo', 'migration');
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'POST /api/v1/repos/repo/migrations',
      'GET /api/v1/repos/repo/migrations/migration'
    ]);
  });
});
